codegen/exe/Main.hs

```diff
@@ -47,6 +46,7 @@ programOptions =
           <> O.help "Output-directory"
           )
 
+main :: IO ()
```

Comment:
are these changes so that you can run this with `failOnAllWarnings`? :D

Reply:
I do not set the options.
But  `nix-build` stops to not use the type-definition.

Reply:
ah. you are right, that was apparently me.

It seems most of the problems right now come down to me not having build `codegen` with nix myself. I just skipped it because the code was already generated ;)